### PR TITLE
Replace ubuntu-latest with onnxruntime-Ubuntu2204-AMD-CPU

### PR DESCRIPTION
**Pull request by jchen351**

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 1]
```diff
@@ -127,7 +127,7 @@ stages:
   jobs:
   - job: D1
     pool:
-      vmImage: ubuntu-latest
+      vmImage: onnxruntime-Ubuntu2204-AMD-CPU
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:130]: This is a pool name. Not an image name.

Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml]: What about ubuntu-22.04? If not, which vmImage do you recommend?

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml]: Should be "name: onnxruntime-Ubuntu2204-AMD-CPU" instead.

**Review by snnn — APPROVED** [review]
Thank you!

**Pull request merged**